packages/core/lib/modules/accounts.js

```diff
@@ -69,28 +132,51 @@ export class Accounts {
     if (existingAccounts.length > 0) {
       throw new Error(Errors.accountAlreadyExists);
     }
+    
+    if (json) {
+      const pair = await KeyringRpc.addFromJson(json, password); 
+      
+      
+      keyPairType = pair.type;
```

Comment:
are we guaranteed a valid response for pair?

Reply:
fixed